code/modules/food_and_drinks/recipes/soup_mixtures.dm

```diff
@@ -167,24 +167,31 @@
 			new_ingredient.reagents?.chem_temp = holder.chem_temp
 			cached_ingredients[new_ref] = new_ingredient.reagents?.total_volume || 1
 
+	var/turf/below_pot = get_turf(pot)
 	for(var/datum/weakref/ingredient_ref as anything in cached_ingredients)
 		var/obj/item/ingredient = ingredient_ref.resolve()
+
 		// An ingredient has gone missing, stop the reaction
 		if(QDELETED(ingredient) || ingredient.loc != holder.my_atom)
 			testing("Soup reaction ended due to having an invalid ingredient present.")
 			return END_REACTION
 
-		// Don't add any more reagents if we've boiled over
-		if(reaction.data["boiled_over"])
-			continue
-
-		// Transfer 20% of the initial reagent volume of the ingredient to the soup
-		transfer_ingredient_reagents(ingredient, holder, max(cached_ingredients[ingredient_ref] * 0.2, 2))
+		// Transfer 20% of the initial reagent volume of the ingredient to the soup.
+		if(!transfer_ingredient_reagents(ingredient, holder, max(cached_ingredients[ingredient_ref] * 0.2, 2)))
+			continue //all reagents were transfered
 
 		// Uh oh we reached the top of the pot, the soup's gonna boil over.
 		if(holder.total_volume >= holder.maximum_volume * 0.95)
-			boil_over(holder)
-			reaction.data["boiled_over"] = TRUE
+			below_pot.visible_message(span_warning("[pot] starts to boil over!"))
+			// Create a spread of dirty foam
```

Comment:
The original intent was to not allow soup reaction to boil over more than once per reaction, as it would get really really annoying.

Reply:
But how does that make sense? If more water/other reagents are added mid reaction it's only realistic to boil over each time. 

Maybe reduce a large amount of reagents when boiling over so it takes more time to get to that stage again?

Reply:
> But how does that make sense?

It was a gameplay abstraction, if it boils over it just does the goofy animation, spashes the chef, and they clean up and move on. If it repeats it'll continuously make foam and *really* mess up people. 

I guess we can cut reagents by like 50% instead

Reply:
Yeah 50% is good. I can do that